homeassistant/components/hunterdouglas_powerview/cover.py

```diff
@@ -179,9 +219,28 @@ async def async_set_cover_position(self, **kwargs):
 
     async def _async_move(self, target_hass_position):
         """Move the shade to a position."""
-        current_hass_position = hd_position_to_hass(self._current_cover_position)
+
+        # we will limit force update to 5 minutes to prevent spamming hub
+        time_delta = datetime.now(tz=None) - timedelta(minutes=FORCED_REFRESH_TIMEFRAME)
+        force_refresh = False
+        if self._last_forced_refresh is not None:
+            force_refresh = self._last_forced_refresh < time_delta
+
+        current_hass_position = self.current_cover_position
         steps_to_move = abs(current_hass_position - target_hass_position)
+
+        if not steps_to_move and force_refresh is True:
+            # force a refresh in case hass was out of date and a move is needed
+            _LOGGER.debug("Cover %s - Force refresh on move", self.name)
+            await self._async_force_refresh_state()
+            current_hass_position = self.current_cover_position
+            steps_to_move = abs(current_hass_position - target_hass_position)
+
+        if not steps_to_move:
+            return
+
```

Comment:
We dropped this check in case the hub was out of sync with the shade so it always sends the command because users reported it wouldn't reliably open/close otherwise
```suggestion
```

Reply:
strange i never noticed that in my syncing with the dev branch - i implented a force refresh on move if the hub hadnt refreshed for 5 minutes for similar reason
Havent had any issues since - force refresh is kind of needed for the tdbu as requiring two different entities means it is not always aware of the other motor being moved